compiler/circle-mpqsolver/src/bisection/BisectionSolver.h

```diff
@@ -0,0 +1,84 @@
+/*
+ * Copyright (c) 2022 Samsung Electronics Co., Ltd. All Rights Reserved
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef __MPQSOLVER_BISECTION_SOLVER_H__
+#define __MPQSOLVER_BISECTION_SOLVER_H__
+
+#include "Quantizer.h"
+#include "Evaluator.h"
+
+#include <luci/IR/Module.h>
+
+#include <memory>
+#include <string>
+
+namespace mpqsolver
+{
+namespace bisection
+{
+
+class BisectionSolver final
+{
+public:
+  /**
+   * @brief options for running bisection algorithm
+   */
+  struct Options
```

Comment:
I guess that "virtual" `Options` came from implementation of CircleQuantizer or CircleOptimizer. That is for implementing multiple versions of Options. BisectionSolver may not require such ability.
(That may be more suitable for more general `MPQSolver`interface)

Could you simplify the option? I think enum class is enough for now.

```
enum Algorithm
{
  Auto, // NYI
  ForceQ16Front,
  ForceQ16Back,
};
```

Ex:
`solver.algorithm(Algorithm::ForceQ16Front);`

Reply:
ok